### PR TITLE
NN-4073: Add new STAFF_SEARCH role for user name search

### PR DESCRIPTION
**Pull request by sp-watson**

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserResource.kt [hunk 1]
```diff
@@ -140,11 +140,11 @@ class UserResource(
   ): StaffDetail =
     userService.findByStaffId(staffId)
 
-  @PreAuthorize("hasRole('ROLE_USE_OF_FORCE')")
+  @PreAuthorize("hasRole('ROLE_USE_OF_FORCE') or hasRole('ROLE_STAFF_SEARCH')")
```
Comment (petergphillips) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserResource.kt:143]: there is a `hasAnyRole` that would probably be better

**Review by petergphillips — DISMISSED** [review]
LGTM

File: src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserResource.kt (hunk 1 above)
Comment (sp-watson) [reply src/main/kotlin/uk/gov/justice/digital/hmpps/nomisuserrolesapi/resource/UserResource.kt:143]: Oh - I'll change it 'cos I can't merge it anyway!

**Review by petergphillips — APPROVED** [review]
LGTM

**Pull request merged**